peer/peer.go

```diff
@@ -1476,6 +1476,7 @@ out:
 				// command.
 				p.PushRejectMsg("malformed", wire.RejectMalformed, errMsg, nil,
 					true)
+				p.Disconnect()
```

Comment:
Hmm, this actually looks to implement the logic which was _described_ in the comment above 🤔